packages/components/src/buttons/src/EmbeddedButton.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+    type StyledComponentProps,
+    useStyledSystem,
+    type ResponsiveProp,
+    useResponsiveValue
+} from "@hopper-ui/styled-system";
+import { type ForwardedRef, forwardRef } from "react";
+import {
+    Button as RACButton,
+    type ButtonProps as RACButtonProps,
+    composeRenderProps,
+    useContextProps
+} from "react-aria-components";
+
+import {
+    composeClassnameRenderProps,
+    cssModule
+} from "../../utils/index.ts";
+
+import { EmbeddedButtonContext } from "./EmbeddedButtonContext.ts";
+
+import styles from "./EmbeddedButton.module.css";
+
+export const GlobalEmbeddedButtonCssSelector = "hop-EmbeddedButton";
+
+export interface EmbeddedButtonProps extends StyledComponentProps<RACButtonProps> {
+    /**
+     * The size of the EmbeddedButton.
+     * @default "md"
+     */
+    size?: ResponsiveProp<"md" | "lg">;
+}
+
+function EmbeddedButton(props: EmbeddedButtonProps, ref: ForwardedRef<HTMLButtonElement>) {
+    [props, ref] = useContextProps(props, ref, EmbeddedButtonContext);
+
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+
+    const {
+        className,
+        size: sizeProp,
+        isDisabled,
+        style: styleProp,
+        children,
+        ...otherProps
+    } = ownProps;
+
+    const size = useResponsiveValue(sizeProp) ?? "md";
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalEmbeddedButtonCssSelector,
+        cssModule(
+            styles,
+            "hop-EmbeddedButton",
+            size
+        ),
+        stylingProps.className
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    return (
+        <RACButton
+            ref={ref}
+            className={classNames}
+            style={style}
+            isDisabled={isDisabled}
+            {...otherProps}
+        >
+            {children}
+        </RACButton>
+    );
+}
+
+/**
+ * EmbeddedButtons are used to initialize an action. EmbeddedButton labels express what action will occur when the user interacts with it.
```

Comment:
what are you referring to when you say EmbeddedButton labels? Is this the best tagline to use here? 
Maybe mention the fact that they're like icon buttons that are used inside other elements.